src/main/java/software/amazon/nio/spi/s3/S3FileSystemProvider.java

```diff
@@ -231,18 +307,24 @@ protected SeekableByteChannel newByteChannel(S3AsyncClient client, Path path, Se
             options = Collections.emptySet();
         }
 
-        final S3Path s3Path = getPath(path.toUri());
+        final S3Path s3Path = (S3Path)path;
         final S3SeekableByteChannel channel;
+        final S3FileSystem fs = s3Path.getFileSystem();
 
-        if (client == null) {
-            client = S3ClientStore.getInstance().getAsyncClientForBucketName(s3Path.bucketName());
-        }
+        channel = new S3SeekableByteChannel(s3Path, fs.client(), options);
+        fs.registerOpenChannel(channel);
 
-        channel = new S3SeekableByteChannel(s3Path, client, options);
-        s3Path.getFileSystem().registerOpenChannel(channel);
         return channel;
     }
 
+    /**
+     * @deprecate -- TODO how to address the deprecation
```

Comment:
As mentioned somewhere, like all other methods that get a S3Client for testing purposing, I am deprecating the method because it is not needed any more and the preferred replacement is by the mean of the client provider. The reason why I would like to get rid of these methods in the future, in addition to make the interface of S3ClientProvider cleaner, is for maintenability purposes (i.e. once we have a protected method it becomes untouchable for backward compatibility).

I forgot to comment the method accordingly to the other:

```
/**
     *
     * @deprecated in favour of using a proper S3ClientProvider in S3FileSystem.
     *             For instance, instead of the following code:
     *             <pre>
     *             S3FileSystemProvider p = ...;
     *             S3Path dir = ...;
     *             S3AsyncClient s3 = ...;
     *             DirectoryStream filter = ...;
     *
     *             p.newDirectoryStream(s3, path, filter);
     *             </pre>
     *             something equivalent to the below should be used:
     *             <pre>
     *             S3FileSystemProvider p = new MyFileSystemProvider(new MyClientProvider());
     *             S3Path dir = ...;
     *
     *             p.newDirectoryStreampath, filter);
     *             </pre>
     */
```